### ISSUE TITLE
Capybara as runtime dependency?

### ISSUE THREAD

**Issue by n-rodriguez** [github/view_component#292]
Hi there!

Do we really need `capybara` as runtime dependency? See : https://github.com/github/view_component/blob/master/view_component.gemspec#L36

It's used in test helper : https://github.com/github/view_component/blob/master/lib/view_component/test_helpers.rb

Thank you!

**Comment by joelhawksley** [comment]
@n-rodriguez only one way to find out! Would you be up for opening a PR to move it to development only?

**Comment by n-rodriguez** [comment]
@joelhawksley done https://github.com/github/view_component/pull/293

**Comment by joelhawksley** [comment]
@n-rodriguez - per @jcoyne's comment here: https://github.com/github/view_component/pull/293#issuecomment-611273636, we need Capybara for the test helpers to work. 

While most people will have it in their apps already, I don't think it makes sense to require folks to install it themselves to use the test helpers.

Was there an issue you ran into that prompted the desire to make this change?

**Comment by n-rodriguez** [comment]
> Was there an issue you ran into that prompted the desire to make this change?

Because it will be installed in production environment. I do use Capybara locally but not in prod :)

**Comment by joelhawksley** [comment]
@n-rodriguez is there is a way to have a gem provide different dependencies for `development` and `prod`? 

From what I understand, `development_dependency` functions to identify dependencies needed for the development _of_ the gem, not using the gem _in_ development/test.

**Comment by n-rodriguez** [comment]
> From what I understand, `development_dependency` functions to identify dependencies needed for the development _of_ the gem, not using the gem _in_ development/test.

Good point! At first I thought you were actually using Capybara in *your* tests but it seems that it's not the case.

> @n-rodriguez is there is a way to have a gem provide different dependencies for `development` and `prod`?

I would say... documentation.

**Comment by rmacklin** [comment]
I agree that it's not ideal to require `capybara` in a production environment - given that it's not used in production, it is essentially a waste of memory to load it.

So personally, I lean towards removing the dependency from `view_component` for a few reasons:
- Many (most?) rails apps will already declare a `test` dependency on `capybara` because [Rails System Tests](https://guides.rubyonrails.org/testing.html#system-testing) depend on it, which means they'll still be able to use the `capybara` assertions in their component tests.
- It's still possible to test components without the `capybara` assertions, by directly using the return value of the `render_inline` helper.
- I don't think it's too much to ask for folks to add `capybara` to their `Gemfile` if they don't already have it and they want to use the `capybara` assertions when testing components.

With that in mind, I've opened a draft PR in https://github.com/github/view_component/pull/298 with my proposal for how to address this.

**Comment by jonspalmer** [comment]
Do we really need TestHelpers to have a dependency on Capybara? Can't we rewrite it to do without? The only place we use it is `refute_component_rendered` which could be stated with some more native assertions. 

Another though is that for folks using something other than Minitest these TestHelpers aren't helping as much. Really just says these helpers should be some other gem.

**Comment by rmacklin** [comment]
> The only place we use it is `refute_component_rendered` which could be stated with some more native assertions.

See https://github.com/github/view_component/pull/216 - I think the reason for including `Capybara::Minitest::Assertions` was to give access to those assertions inside instances of `ViewComponent::TestCase` (those methods leverage the `page` definition we supply). The `refute_component_rendered` helper was only added later in https://github.com/github/view_component/pull/263.

**Comment by jonspalmer** [comment]
@rmacklin makes sense in the Minitest case. How much does this logic help if you're using RSpec or some other testing library? The testing pattern if very different there and you wouldn't be inheriting from `ViewComponent::TestCase` and would be requiring `capybara/rspec` yourself.

**Comment by joelhawksley** [comment]
Closed by https://github.com/github/view_component/pull/298

**Comment by rmacklin** [comment]
@jonspalmer I think you're right that things are a bit murky w.r.t. rspec. We're including capybara's minitest assertions module in `ViewComponent::TestHelpers`:
https://github.com/github/view_component/blob/d3952cb35c82734b5119bc2c173e99cc08219d5c/lib/view_component/test_helpers.rb#L7
but we also have documentation suggesting that `ViewComponent::TestHelpers` should be included in rspec `component` configuration (where it seems like the minitest assertions would be out of place):
https://github.com/github/view_component/blob/dc0c0c1671dc116a3c2b822a5031b4045c2eb569/README.md#L553-L562
That said, there definitely are consumers of this library that use rspec and I haven't seen them raise an issue about this being a problem so far. I suspect they are just not using those methods.

I don't have a lot of rspec experience, but I also noticed that the [capybara README section on rspec](https://github.com/teamcapybara/capybara/blob/3.32.1/README.md#using-capybara-with-rspec) mentions:
> Capybara matchers are also supported in view specs

I wonder if folks have been testing components in [view specs](https://relishapp.com/rspec/rspec-rails/v/4-0/docs/view-specs/view-spec) where they can already leverage capybara's rspec matchers.

Anyway, might be better to open a new issue to discuss the rspec stuff since this one about removing the capybara dependency has been closed.
